include/seqan3/range/views/minimiser_hash.hpp

```diff
@@ -0,0 +1,158 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Mitra Darvish <mitra.darvish AT fu-berlin.de>
+ * \brief Provides seqan3::views::minimiser_hash.
+ */
+
+#pragma once
+
+#include <seqan3/range/views/kmer_hash.hpp>
+#include <seqan3/range/views/minimiser.hpp>
+
+namespace seqan3::detail
+{
+//!\brief views::minimiser_hash's range adaptor object type (non-closure).
+struct minimiser_hash_fn
+{
+    /*!\brief Store the shape and the window size and return a range adaptor closure object.
+    * \param[in] shape       The seqan3::shape to use for hashing.
+    * \param[in] window_size The windows size to use.
+    * \throws std::invalid_argument if the size of the shape is greater than the `window_size`.
+    * \returns               A range of converted elements.
+    */
+    constexpr auto operator()(shape const & shape, uint32_t const window_size) const
+    {
+        return seqan3::detail::adaptor_from_functor{*this, shape, window_size};
+    }
+
+    /*!\brief Store the shape, the window size and the seed and return a range adaptor closure object.
+    * \param[in] shape       The seqan3::shape to use for hashing.
+    * \param[in] window_size The size of the window.
+    * \param[in] seed        The seed to use.
+    * \throws std::invalid_argument if the size of the shape is greater than the `window_size`.
+    * \returns               A range of converted elements.
+    */
+    constexpr auto operator()(shape const & shape, uint32_t const window_size, uint64_t const seed) const
```

Comment:
hmm usually we try to avoid having to size_t's in a constructor because you can easily mix up the order..
That's why we usually introduce strong types. I think this is necessary here too. Sorry for the additional work :/ Do you know how to introduce a strong_type ? We have a `detail::strong_type` base class for this.